classes/class-wpcom-liveblog-wp-cli.php

```diff
@@ -13,67 +13,96 @@ public function readme_for_github() {
 									) );
 
-		foreach( $posts->posts as $post ) {
+		//How many live blogs do we have?
+		$total_liveblogs = count($posts->posts);
+		$current_liveblog = 0;
+
+		//Feedback to the user
+		WP_CLI::line( 'Found ' . $total_liveblogs . ' Live Blogs.' );
+
+		foreach ($posts->posts as $post) {
+
+			//Increment the count so we get a more human readable index, inital value becomes 1 rather than 0.
+			$current_liveblog ++;
+
+			//Tell the user what we are doing, but lets colour this one se we can see its a new Liveblog in the console output.
+			// Note: Assigned to a varibale to be echo'd out, calling WP_CLI::colorize directly didnt output.
+			$intro = WP_CLI::colorize( "%4 Processing Liveblog {$current_liveblog} of {$total_liveblogs} %n");
+			echo $intro . PHP_EOL;
+
+		    //Define the post ID
 			$post_id = $post->ID;
 
 			// get all entries that have been edited in the liveblog
 			$entries_query = new WPCOM_Liveblog_Entry_Query( $post_id, WPCOM_Liveblog::key );
-			$edit_entries = $entries_query->get_all_edits( array( 'post_id' => $post_id ) );			
+			$edit_entries = $entries_query->get_all_edits( array( 'post_id' => $post_id ) );
+
 			// find correct comment_ids to replace incorrect meta_values
 			$correct_ids_array = $wpdb->get_results(
 				$wpdb->prepare(
 				"SELECT comment_id FROM $wpdb->comments
-				 WHERE comment_post_id = %d AND comment_id NOT IN 
+				 WHERE comment_post_id = %d AND comment_id NOT IN
 				 ( SELECT $wpdb->commentmeta.comment_id FROM $wpdb->commentmeta
-				   INNER JOIN $wpdb->comments 
+				   INNER JOIN $wpdb->comments
 				   ON $wpdb->comments.comment_id = $wpdb->commentmeta.comment_id
 				   WHERE comment_post_id = %d )
 				 ORDER BY comment_id ASC", $post_id, $post_id )
 				);
 			$correct_ids = wp_list_pluck( $correct_ids_array, 'comment_id' );
 
 			// replace incorrect meta_value with correct one
-			foreach( $edit_entries as $edit_entry ) {
-				$entry_id = $edit_entry->get_id();
+			if( count( $edit_entries ) > 0 ){
+				foreach( $edit_entries as $edit_entry ) {
+					$entry_id = $edit_entry->get_id();
 
-				// look for replaces property in $correct_ids 
-				if ( in_array( $edit_entry->replaces, $correct_ids ) ) {
-					continue;
-				}
-				else {
-					for ( $i = 0; $i <= count( $correct_ids ) - 1; $i++ )
-					{		
-						// replace with correct meta_value
-						if ( $correct_ids[$i] < $entry_id ) {
-
-							$wpdb->update(
-								$wpdb->commentmeta,
-								array( 'meta_value' => $correct_ids[$i] ),
-								array( 'comment_id' => $entry_id )
-								);
+					// look for replaces property in $correct_ids
+					if ( in_array( $edit_entry->replaces, $correct_ids ) ) {
+
+						continue;
+
+					} elseif( false == $is_dryrun ) {
+						// If this isnt a dry run we can run the database Update.
+						for ( $i = 0; $i <= count( $correct_ids ) - 1; $i++ ) {
+
+							// replace with correct meta_value
+							if ( $correct_ids[$i] < $entry_id ) {
+
+								$wpdb->update(
```

Comment:
Let's move the dry run check from above to here, so we can report in the dry run on what IDs are changing.